### PR TITLE
feat: allow help to be shown from anywhere

### LINKED ISSUE #376: Allow viewing usage description with `-h` when outside a Melos workspace
Currently, Melos always reports an error when invoked outside a Melos workspace. To explore Melos commands, it's useful to be able to view the usage description outside a Melos workspace.

### PR DESCRIPTION
**Pull request by SscSPs**
Closes #376

Signed-off-by: Sahil Soni <SscSPs@gmail.com>

## Description
Allow help to be run from anywhere and not just from inside a melos workspace

## Type of Change

- [x] ✨ `feat` -- New feature (non-breaking change which adds functionality)
- [ ] 🛠️ `fix` -- Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ `!` -- Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 `refactor` -- Code refactor
- [ ] ✅ `ci` -- Build configuration change
- [ ] 📝 `docs` -- Documentation
- [ ] 🗑️ `chore` -- Chore

### CODE REVIEW

**Comment by SscSPs** [comment]
Can you please review this PR
And, does this count under hacktoberFest? just making sure, I think it should, but not sure, added the closes tag with a hacktoberfest issue, do i need to add something else as well ?

File: packages/melos/bin/melos.dart [hunk 1]
```diff
@@ -40,7 +40,13 @@ Future<void> main(List<String> arguments) async {
     return;
   }
   try {
-    final config = await MelosWorkspaceConfig.fromDirectory(Directory.current);
+    var allowOutside = false;
+    if (arguments.contains('--help') || arguments.contains('-h')) {
+      allowOutside = true;
+    }
```
Comment (blaugold) [inline packages/melos/bin/melos.dart:46]: This can just be:

```dart
final hasHelpOption = arguments.contains('--help') || arguments.contains('-h');
```

**Review by blaugold — CHANGES_REQUESTED** [review]
Thanks for this contribution!

LGTM. Just one nit.

File: packages/melos/bin/melos.dart (hunk 1 above)
Comment (SscSPs) [reply packages/melos/bin/melos.dart:46]: The reason i put it this way was so if in future we want more flags to be available outside the melos workspaces, we can update this condition here.

Comment (blaugold) [reply packages/melos/bin/melos.dart:46]: How about this, then:
```dart
final hasHelpOption = arguments.contains('--help') || arguments.contains('-h');
final useEmptyConfig = hasHelpOption;
```
We avoid a mutable variable, it's shorter. If we later want to run with an empty config in other situations, we can add `|| <condition>` behind `hasHelpOption` and have an easily readable list of conditions when we need to use an empty config.

Comment (SscSPs) [reply packages/melos/bin/melos.dart:46]: Makes sense, may be as well just move it out to a function, 
```
bool shouldUseEmptyConfig(List<String> arguments)
```
And move all this there and use that

Comment (blaugold) [reply packages/melos/bin/melos.dart:46]: That sounds good.

Comment (SscSPs) [reply packages/melos/bin/melos.dart:46]: Cool, will make the changes tomorrow, busy today

Comment (SscSPs) [reply packages/melos/bin/melos.dart]: made the changes, let me know if there's something else needed as well :)

**Review by blaugold — APPROVED** [review]

**Pull request merged**